### PR TITLE
Add Basic Geofencing and Significant Location Monitoring Support, Ability to Receive didChangeAuthorization callbacks, Additional Stubbed Methods

### PR DESCRIPTION
**Pull request by quicklywilliam**
Also cleaned up file structure a bit and re-added support for iOS 8 devices

### CODE REVIEW

File: Source/LocationManager.swift [hunk 1]
```diff
@@ -18,6 +18,14 @@ open class LocationManager {
             self = .coreLocation
         }
     }
+    open func authorizationStatus() -> CLAuthorizationStatus {
```
Comment (quicklywilliam) [inline Source/LocationManager.swift]: Adding this as a class method means that to use LocationManager you need to switch over calls to authorizationStatus to the instance. This worked for my use-case, but it isn't ideal.

**Comment by vermont42** [comment]
Thanks for the PR! May I ask how you are using `GpxLocationManager`? Testing or development of `Ride Report`, perhaps? I am delighted to see others using something I created.

**Comment by quicklywilliam** [comment]
You are quite welcome @vermont42! I am indeed using it to test Ride Report. I've also got some associated tools I've built for unit and UI testing Ride Report. I think they could be useful in other location-driven apps, perhaps some of these would be appropriate to add to GpxLocationManager?

**Comment by vermont42** [comment]
I'm open to unit and UI tests. Would the UI tests potentially drive the demo app? How do you quickly unit test `GpxLocationManager` when the passage of time is so central to its functionality?

**Comment by quicklywilliam** [comment]
So far I've been doing a mixture of automated units tests and using GpxLocationManager to drive manual UI testing. Without getting into a large digression on the value of automated UI testing, we have chosen not to do it (for now) and focus instead on our unit tests. That said, even for manual UI testing, GpxLocationManager has enormous value because we can do it from our chairs instead of having to go for a bike ride (much as we'd love an excuse to do just that).

For both the unit and UI testing, we're taking advantage of GpxLocationManager's secondLength property. I've had to do a bit of fixing both in my code and GpxLocationManager to get it all working right, but the result is that we can run our tests at at least 10x real time without breaking anything.

**Pull request merged**